algorithms/a3c/main.py

```diff
@@ -40,80 +54,275 @@
+                    help='number of episodes passed for resuming')
+parser.add_argument('--verbose-num-steps', default=False,
+                    help='print step number every args.num_steps')
+parser.add_argument('--num-steps', type=int, default=20,
+                    help='number of forward steps in A3C (default: 20)')
+
+# ai2thor settings
+parser.add_argument('--config-file-name', default='default_config.json',
+                    help='File must be in gym_ai2thor/config_files. Other good variants with '
+                         'natural language are: NL_pickup_5_objects_surround_fp_211_v0.1.json and'
+                         'NL_pickup_multiple_cups_only_fp404_v0.1.json')
+parser.add_argument('--resume-latest-config', default=1,
+                    help='Whether to resume latest_config.json found in experiment folder'
+                         'Default is set so user cannot override settings from text file')
+
+parser.add_argument('--max-episode-length', type=int, default=1000,
+                    help='maximum length of an episode (default: 1000000)')
+parser.add_argument('--num-random-actions-at-init', type=int, default=0,
+                    help='Number of random actions the agent does on initialisation')
+
 # Atari arguments. Good example of keeping code modular and allowing algorithms to run everywhere
 parser.add_argument('--atari', dest='atari', action='store_true',
                     help='Run atari env instead with name below instead of ai2thor')
 parser.add_argument('--atari-render', dest='atari_render', action='store_true',
                     help='Render atari')
 parser.add_argument('--atari-env-name', default='PongDeterministic-v4',
                     help='environment to train on (default: PongDeterministic-v4)')
-#
 parser.set_defaults(atari=False)
 parser.set_defaults(atari_render=False)
 
+# VizDoom arguments
+parser.add_argument('-d', '--difficulty', type=str, default="hard",
+                    help="""Difficulty of the environment,
+                    "easy", "medium" or "hard" (default: hard)""")
+parser.add_argument('--living-reward', type=float, default=0,
+                    help="""Default reward at each time step (default: 0,
+                    change to -0.005 to encourage shorter paths)""")
+parser.add_argument('--frame-width', type=int, default=300,
+                    help='Frame width (default: 300)')
+parser.add_argument('--frame-height', type=int, default=168,
+                    help='Frame height (default: 168)')
+parser.add_argument('-v', '--visualize', type=int, default=0,
+                    help="""Visualize the envrionment (default: 0,
+                    use 0 for faster training)""")
+parser.add_argument('--sleep', type=float, default=0,
+                    help="""Sleep between frames for better
+                    visualization (default: 0)""")
+parser.add_argument('--vizdoom', dest='vizdoom', action='store_true',
```

Comment:
What I did with rainbow was to let people select "game", so the argument is a string saying "ai2thor", "space-invaders" or any of the many environments out there. I suggest you use a similar parameter here instead of one per environment